### PR TITLE
UHF-5253: Job search location filter

### PR DESCRIPTION
**Pull request by xkhaven**
# [UHF-5253](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5253)

## What was done

* Added area selection for filtering search result

## How to install

* Make sure your instance is up and running on latest dev branch.
  * Use instance of https://github.com/City-of-Helsinki/drupal-helfi-rekry
  * `git pull origin dev`
  * `make fresh`
  * `make shell`
  * `drush sapi-c; drush sapi-rt; drush sapi-i; drush cr`
* Update the HDBT theme
  * `composer require drupal/hdbt:dev-UHF-5253_Job_location_filter`
* Run `make drush-cr drush-locale-update`

## How to test

* [ ] Open job search: https://helfi-rekry.docker.so/fi/avoimet-tyopaikat/etsi-avoimia-tyopaikkoja
* [ ] Form item Job locations has dropdown with all city areas
* [ ] Search results are filtered by the selection
* [ ] Check that code follows our standards

## Designers review

* [X] This PR does not need designers review

## Other PRs

* https://github.com/City-of-Helsinki/drupal-helfi-rekry/pull/380

[UHF-5253]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-5253?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Jussiles — CHANGES_REQUESTED** [review]
Good work! Some thing that should be fixed before the merge.

**Review by Jussiles — APPROVED** [review]
it's good now! 🦜

File: src/js/react/apps/job-search/enum/PostalCodes.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+
```
[the remark this answers is not in the document]
Comment (xkhaven) [reply src/js/react/apps/job-search/enum/PostalCodes.ts]: Removed

File: src/js/react/apps/job-search/containers/FormContainer.tsx [hunk 2]
```diff
@@ -189,28 +197,47 @@ const FormContainer = () => {
...
-                )}
-                placeholder={Drupal.t('All languages', {}, { context: 'Language placeholder' })}
-                label={languageLabel}
-                // @ts-ignore
-                options={languagesOptions}
-                value={languageSelection}
-                id={SearchComponents.LANGUAGE}
-                onChange={setLanguageFilter}
-              />
+            <div className='job-search-form__dropdowns__upper'>
```
[the remark this answers is not in the document]
Comment (xkhaven) [reply src/js/react/apps/job-search/containers/FormContainer.tsx]: Corrected.

File: src/js/react/apps/job-search/store.tsx [hunk 3]
```diff
@@ -299,3 +301,13 @@ export const resetFormAtom = atom(null, (get, set) => {
   set(urlUpdateAtom, {});
   set(languageSelectionAtom, null);
 });
+
+export const areaFilterAtom = atom(async (get) => getAreaInfo.map((item: any) => (
```
[the remark this answers is not in the document]
Comment (xkhaven) [reply src/js/react/apps/job-search/store.tsx]: Changed.

**Pull request merged**